app/commands/payments/payment/create.rb

```diff
@@ -0,0 +1,33 @@
+# This responds to a new payment that has been made
+# and creates a record of it in our database. The actual
+# creation of the payment within Stripe happens through
+# "payment intents".
+
+class Payments::Payment::Create
+  include Mandate
+
+  initialize_with :user, :provider, :product, :external_id, :amount_in_cents, :external_receipt_url, subscription: nil
+
+  def call
+    Payments::Payment.create!(
+      user:,
+      provider:,
+      product:,
+      external_id:,
+      external_receipt_url:,
+      subscription:,
+      amount_in_cents:
+    ).tap do |payment|
+      User::UpdateTotalDonatedInCents.(user) if product == :donation
```

Comment:
This assumes that payments for premium don't count towards one's total donation amount